_posts/2025-03-26-phased-strimzi-upgrade-example.md

```diff
@@ -0,0 +1,700 @@
+It will then roll the cluster, one broker at a time, on to the new image.
+
+```mermaid
+flowchart TD
+  S_043(Strimzi 0.43) --> K_A(Kafka A)
+  S_043 --> K_B(Kafka B)
+	S_044(Strimzi 0.44) --> K_C(Kafka C)
+```
+
+After a burn-in period, where we are happy with the stability of the cluster, our client side canary jobs are stable and no client teams are reporting issues, we begin to move the rest of the fleet.
+
+```mermaid
+flowchart TD
+	S_043(Strimzi 0.43)
+  S_044(Strimzi 0.44) --> K_A(Kafka A)
+  S_044 --> K_B(Kafka B)
+	S_044 --> K_C(Kafka C)
+	style K_A stroke:#f66,stroke-width:2px,stroke-dasharray: 5 5
+```
+
+All is well until we move our highest tier cluster, A. 
+The move to Strimzi 0.44 revealed a configuration issue.
+We think we know what the issue is and how to fix it, but as this is a critical cluster we need to roll back and make sure all the dependant services are running correctly. 
+In this setup the roll back is a simple label change:
+
+```shell
+kubectl -n kafka label --overwrite kafka kafka-a strimzi-resource-selector=strimzi-0-43-0
+```
+
+```mermaid
+flowchart TD
+  S_043(Strimzi 0.43) --> K_A(Kafka A)
+  S_044(Strimzi 0.44) --> K_B(Kafka B)
+	S_044 --> K_C(Kafka C)
+```
+
+We fix the configuration issue and move Kafka cluster A over to the 0.44 operator.
+
+```mermaid
+flowchart TD
+  S_043(Strimzi 0.43)
+	style S_043 stroke:#f66,stroke-width:2px
+  S_044(Strimzi 0.44) --> K_A(Kafka A)
+  S_044 --> K_B(Kafka B)
+	S_044 --> K_C(Kafka C)
+```
+
+At this point the Strimzi 0.43 operator is not managing any Kafka clusters and all the clusters being managed by the 0.44 operator have been stable for sufficient period that we are confident we don't need to roll back. 
+We can now remove the Strimzi 0.43 operator deployment and its associated RBAC resources.
+At this point we can add a new version to the Operator Set, adding the 0.45 installation files and upgrading the CRDs to those of the 0.45 version.
+
+```mermaid
+flowchart TD
+  S_044(Strimzi 0.44) --> K_A(Kafka A)
+  S_044 --> K_B(Kafka B)
+	S_044 --> K_C(Kafka C)
+  S_045(Strimzi 0.45)
+	style S_045 stroke:#008000,stroke-width:2px
+```
+
+#### Things to watch out for
+
+The main things to watch for when using this Strimzi deployment pattern are:
+- Take note of the advice in the [CRD section](#custom-resource-definitions-1) above around the number of concurrent versions and CRD API version upgrades:
+  - The only supported configuration is running two adjacent Strimzi versions concurrently.
+  - When upgrading the CRD API version. You will need to upgrade all operands to the last Strimzi version before the API upgrade, before adding the new Strimzi version with the new CRD API version.
+- Extreme care should be taken, while updating the Operator Set, to not remove the CRDs from the Kubernetes cluster. Doing so would result in the removal of **all** Kafka clusters (and other Strimzi operands).
```

Comment:
```suggestion
- Take extreme care when updating the Operator Set to avoid removing the CRDs from the Kubernetes cluster. Doing so would delete all Kafka clusters and other Strimzi-managed resources.
```